src/components/ProgramCard.tsx

```diff
@@ -0,0 +1,161 @@
+import React from "react";
+import {
+    Box,
+    Wrap,
+    WrapItem,
+    Badge,
+    Center,
+    AspectRatio,
+    Image,
+} from "@chakra-ui/react";
+import type { ProgramCardInfo } from "models/Program";
+import { programFormat } from "@prisma/client";
+
+type ProgramCardProps = {
+    styleProps?: Record<string, unknown>;
+    cardInfo: ProgramCardInfo[];
+    session: Record<string, unknown>;
+};
+
+/**
+ *
+ * @param cardInfo info for the program cards on the home page
+ * @returns a component with all the cards for each of the programs in the database
+ */
+export const ProgramCard: React.FC<ProgramCardProps> = ({
+    cardInfo,
+    session,
+}): JSX.Element => {
+    // For spacing, when the #cards in a row < max #cards
+    const placeholders: ProgramCardInfo[] = [
+        {
+            id: 0,
+            name: "placeholder",
+            image: "",
+            description: "",
+            startDate: new Date(),
+            endDate: new Date(),
+            onlineFormat: programFormat.online,
+            tag: "",
+            price: 0,
+        },
+        {
+            id: 0,
+            name: "placeholder",
+            image: "",
+            description: "",
+            startDate: new Date(),
+            endDate: new Date(),
+            onlineFormat: programFormat.online,
+            tag: "",
+            price: 0,
+        },
+    ];
+
+    return (
+        <Center>
+            <Wrap spacing="50px" justify="space-between">
+                {cardInfo.concat(placeholders).map((item, idx) => {
+                    return item.name == "placeholder" ? (
+                        <WrapItem
+                            flexBasis="300px"
+                            flexGrow={1}
+                            display="hidden"
+                            key={idx}
+                        ></WrapItem>
+                    ) : (
+                        <WrapItem flexBasis="300px" key={idx} flexGrow={1}>
+                            <Box
+                                borderWidth="1px"
+                                width="100%"
+                                _hover={{
+                                    borderColor: "#0C53A0",
+                                    borderWidth: 1,
+                                }}
+                                onClick={() => {
+                                    if (!session)
+                                        window.location.href = "/login";
```

Comment:
How come we aren't using next/router for handling navigation?

https://nextjs.org/docs/api-reference/next/router

Reply:
Add TODO